Joss changes

ilikecubesnstuff opened the pull request:
Implement changes specified under openjournals/joss-reviews#7009.

ilikecubesnstuff committed:
:arrow_up: Support Astropy v6.
Resolves #9.

ilikecubesnstuff committed:
:sparkles: Allow specification of integration method arguments.
Resolves #7.

ilikecubesnstuff committed:
:memo: Add relevant documentation.
Resolves #2, resolves #3, resolves #4, resolves #5, resolves #6.

ilikecubesnstuff committed:
:adhesive_bandage: Allow pattern speed specification in `get_points`.

ilikecubesnstuff committed:
:white_check_mark: Import subpackages from gala explicitly.

ilikecubesnstuff committed:
:arrow_down: Downgrade astropy version for gala testing.

ilikecubesnstuff committed:
:bug: Make `ORBIT_TYPE` a property in all backends.
:white_check_mark: Add test cases to address #7.
:white_check_mark: Add tests for `get_points`.

ilikecubesnstuff committed:
:memo: Specify how to define new backends.
Related to ilikecubesnstuff/commensurability#20.

michael-petersen commented on src/pidgey/agama_backend.py:
```diff
@@ -11,11 +11,18 @@ def __imports__():
+        pot,
+        dt,
+        steps,
+        pattern_speed=0 * u.km / u.s / u.kpc,
```
Now that I'm looking at this again: I wonder if `pattern_speed` makes more sense as an `integration_kwarg`? Promoting to a named argument makes sense for our bar-driven application, but since the tool is more flexible it might give a sense of importance that wasn't intended.

ilikecubesnstuff replied:
This makes sense, but the pattern speed is currently used for `get_points` in some cases as well. Moving it into `integration_kwargs` and out of `get_points` might be better implemented in a future version.

ilikecubesnstuff committed:
:white_check_mark: Fix Agama testing to a particular commit.
:white_check_mark: Comment out test for specifying the integrator, since that is only supported in later Agama versions.

ilikecubesnstuff committed:
:memo: Add external installation links to README.md.
:memo: Specify supported `agama` commit hash.
